blocks/library/heading/index.js

```diff
@@ -151,13 +151,17 @@ registerBlockType( 'core/heading', {
 				onFocus={ setFocus }
 				onChange={ ( value ) => setAttributes( { content: value } ) }
 				onMerge={ mergeBlocks }
-				onSplit={ ( before, after, ...blocks ) => {
-					setAttributes( { content: before } );
-					insertBlocksAfter( [
-						...blocks,
-						createBlock( 'core/paragraph', { content: after } ),
-					] );
-				} }
+				onSplit={
+					insertBlocksAfter ?
```

Comment:
Should this be the responsibility of the block implementer to account for? Or could we just turn `insertBlocksAfter` and `setAttributes` into `noop` for a locked editor?

Reply:
Mmm, I didn't think about it. Sounds reasonable